### DISCUSSION TITLE
Linguagem de programação em desenvolvimento Mobile pra quem está começando.

### DISCUSSION THREAD

**Discussion by Emerson-developer** [frontendbr/forum#1723] [Discussão]
Bom pessoal, ingressei este ano no curso de Engenharia de Software no Instituto Federal,  e lá o professor decidiu ensinar toda parte de lógica de programação em C pois vamos precisar quando chegar em Estrutura de Dados, mas ainda estou em duvida em qual linguagem de programação eu vou me especializar. Estava pensando em aprender C++ para aproveitar esse estudo em C. Mas ainda não tenho certeza se é a melhor opção tendo em vista que quero focar em desenvolvimento mobile, estava pensando em aprender as tecnologias do JavaScript também. 

//Não me atentei a ver se podia fazer isso por aqui, mas aceito ser "adotado" para aprender.

**Comment by theandersonn** [comment]
@SuvinoCoding 

**Uma dica:** 

Tenha paciência, foque em aprender o que está sendo ensinado em sua grade, vai parecer chato, que não vai te levar a lugar nenhum, ninguém usa Linguagem C...

Mas aprender: Lógica de programação / C / Estrutura de Dados / Orientação à objetos 
Vai lhe capacitar e ajudar a ser um programador maduro e trabalhar com linguagens diferentes.

**Sobre Mobile:**

Pra lhe ajudar que direção seguir, existem as aplicações nativas iOS / Android e as soluções híbridas como React Native e Flutter.  Tem demandas para as duas direções, e estamos num momento em que o mercado está aquecido para as soluções híbridas.

Pesquise as especifidades de desenvolver nativamente.
Pesquise como funcionam as soluções híbridas.

Já conhece a linguagem JavaScript?

**Comment by Emerson-developer** [comment]
@theandersonn Sim, foi a primeira linguagem que me interessei mas vi que ela é mais voltada para o desenvolvimento Web (que seria super interessante aprender também para fazer um portifólio legal). Então estava em duvida se valia a pena aprender tecnologias como React Native e NodeJS para usar a linguagem em Mobile, por isso pensei em aprender outra linguagem mais específica.

**Comment by theandersonn** [comment]
O JavaScript é a base do React Native. Ou seja é um framework feito em JavaScript. 
Primeiro temos que aprender JS e em sequencia aprender as características de determinado framework, esse fluxo não nos engessa com determinada ferramenta.
